Formula/simdutf.rb

```diff
@@ -18,9 +18,12 @@ class Simdutf < Formula
   depends_on "cmake" => :build
   depends_on "python@3.11" => :build
   depends_on "icu4c"
-  depends_on "libiconv"
   depends_on macos: :catalina
 
+  on_macos do
+    depends_on "libiconv"
+  end
```

Comment:
Seems weird to even be using the formula at all.

Reply:
I tried locally and system `libiconv` causes some issue when linking Homebrew's `icu4c`.

However, I think `icu4c` is only used when building benchmark program (which also has a requirement on building tests).

Could disable benchmark+testing, but will need a new test as `simdutf` is running benchmark.

Reply:
Works completely fine for me locally though. Weird.

Reply:
May actually be a local issue when testing via Rosetta as it seems to find my ARM libraries. Can try dropping `libiconv`. Still need to figure out Linux test failure.

Reply:
Removing it seems to work fine at #123171.